app/models/credential.rb

```diff
@@ -7,33 +7,56 @@ class Credential < ApplicationRecord
 
   before_create :set_expires_at
 
-  def logged_in?
-    jwt_payload["login"].present?
-  rescue JWT::ExpiredSignature
+  def admin?
+    return true if Rails.env.development?
+
+    if @admin.nil?
```

Comment:
the instance variable exists to let me create admin credentials using the factory, otherwise it is unused.

Reply:
I'd prefer the implementation doesn't add code to test solely the test env. Instead we could use the factory traits to modify the underlying `client` instance variable, as this is the interface that modifies this class behaviour.
https://github.com/zooniverse/caesar/blob/d08f099ac5d64e920f7253ba82cad48d04db8a33/app/models/credential.rb#L90

You can stub the client behaviour for each case, similar to how marten approached it originally https://github.com/zooniverse/caesar/blob/083f496ef14f116e1e159a39b3b3c9c4a6ecdc6a/spec/factories/credential_factory.rb#L14

E.g. `authenticated_admin?`
```
# add whatever else you need to get it working, maybe token_expiry?
credential.instance_variable_set(:@client, {"authenticated_admin?" => evaluator.admin})
```

Reply:
I don't see that this approach allows me to stub methods, however. For some reason trying to include `rspec/rails` in the factory file, like i'd need to make an instance_double, causes an error deep in the middleware about frozen arrays. I suppose I could use ostruct but that seems messy?

Reply:
My main concern with this implementation is we don't test the production code paths, i.e. the next line `client.authenticated_admin?` could be mispelt / incorrect, missing etc and our specs would always pass, production release code would fail. 

I've offered a slightly different solution in https://github.com/amyrebecca/caesar/pull/1

Reply:
I really didn't want to go this route since it involves replacing every single call to `build :credential` with a call to `fake_credential`

Reply:
but i guess i have time to go through and do that.